vmm/src/main.rs

```diff
@@ -44,7 +54,7 @@ fn main() -> ExitCode {
 
     let slice = unsafe { core::slice::from_raw_parts_mut(load_addr, mem_size) };
 
-    let allocator = BuddyAllocator::new(slice);
+    let allocator: &BuddyAllocator = Box::leak(Box::new(BuddyAllocator::new(slice)));
```

Comment:
This doesn't need to be boxed and leaked if you use `thread::scope` to force the thread to live shorter than `'static`, and you clone the allocator to pass it into the thread (`BuddyAllocator` impls `Clone`).